Bottom: 255fe58f78520c88e027156b411f26af103c8f8d
Top:    3cd4653fce4a0100eda2486689c3863dff040a16
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-06 00:33:33 +0100

Refresh of hid-gtk-connect-up-the-file-ch

---

diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index b14aa4c..d72be88 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -405,18 +405,12 @@ ghid_sync_with_new_layout (void)
 }
 
 void
-ghid_notify_save_pcb (const char *file, bool done)
+ghid_notify_save_pcb (const char *filename, bool done)
 {
   /* Do nothing if it is not the active PCB file we're watching
    * that is being saved.
-   *
-   * Ideally, the core ought to notify us for a "SaveAs" type action
-   * so we could re-wire our file-monitor, but it doesn't. This works
-   * however, as the core sets the new PCB file-name before it saves,
-   * and this notification causes us to disconnect our old file-monitor
-   * (pointing at the old file), then we re-connect to the new file.
    */
-  if (strcmp (file, PCB->Filename) != 0)
+  if (strcmp (filename, PCB->Filename) != 0)
     return;
 
   if (!done)
@@ -425,6 +419,14 @@ ghid_notify_save_pcb (const char *file, bool done)
     connect_file_change_monitor (ghidgui);
 }
 
+void
+ghid_notify_pcb_filename_changed (const char *old_filename,
+                                  const char *new_filename)
+{
+  disconnect_file_change_monitor (ghidgui);
+  connect_file_change_monitor (ghidgui);
+}
+
 /* ---------------------------------------------------------------------------
  *
  * layer_process()
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 3a37b66..1ec824c 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -339,6 +339,7 @@ int ghid_drc_window_throw_dialog (void);
 /* In gui-top-window.c  */
 void ghid_update_toggle_flags (void);
 void ghid_notify_save_pcb (const char *file, bool done);
+void ghid_noiify_pcb_filename_changed (const char *old_filename, const char *new_filename);
 void ghid_install_accel_groups (GtkWindow *window, GhidGui *gui);
 void ghid_remove_accel_groups (GtkWindow *window, GhidGui *gui);
